src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -496,31 +498,46 @@ export class KernelDebugAdapter implements DebugAdapter, IKernelDebugAdapter, ID
         await this.dumpCell(cell.document.uri.toString());
 
         if (this.configuration.__mode === KernelDebugMode.RunByLine) {
-            const initialBreakpoint: DebugProtocol.SourceBreakpoint = {
-                line: 1
-            };
-            const splitPath = cell.notebook.uri.path.split('/');
-            const name = splitPath[splitPath.length - 1];
-            const message: DebugProtocol.SetBreakpointsRequest = {
-                seq: seq + 1,
-                type: 'request',
-                command: 'setBreakpoints',
-                arguments: {
-                    source: {
-                        name: name,
-                        path: cell.document.uri.toString()
-                    },
-                    lines: [1],
-                    breakpoints: [initialBreakpoint],
-                    sourceModified: false
+            const textLines = cell.document.getText().splitLines({ trim: false, removeEmptyEntries: false });
```

Comment:
Can you add a comment here explaining why we need to parseForComments? I assume this is because before the breakpoint would end up on a commented line and not work.

Reply:
that's it, will do